onboarding/source/amb_light_service.c

```diff
@@ -18,22 +28,99 @@ static void lightServiceTask(void * pvParameters);
 uint8_t initLightService(void) {
     /* USER CODE BEGIN */
     // Create the task and queue here.
+    BaseType_t xReturned = pdFAIL;
+    if(lightServiceTaskHandle == NULL){
+        xReturned = xTaskCreate(lightServiceTask,
+                                LIGHT_SERVICE_NAME,
+                                LIGHT_SERVICE_STACK_SIZE,
+                                NULL,
+                                LIGHT_SERVICE_PRIORITY,
+                                &lightServiceTaskHandle);
 
+    }
+
+    if(xReturned == pdFAIL){
+        return 0;
+    }
+
+    
+    
+    if(xLightServiceQueue == NULL){
+	    
+	xLightServiceQueue = xQueueCreate(LIGHT_SERVICE_QUEUE_SiZE, LIGHT_EVENT_SIZE);
+	    
+	if(xLightServiceQueue == NULL){
+        	return 0;
+	}
+    }
     /* USER CODE END */
     return 1;
 }
 
+uint16_t getLightSensorData(void)
+{
+
+	adcData_t lightData;
+ 	/** - Start Group1 ADC Conversion 
+ 	*     Select Channel 6 - Light Sensor for Conversion
+ 	*/
+	adcStartConversion(adcREG1, adcGROUP1);
+
+ 	/** - Wait for ADC Group1 conversion to complete */
+ 	while(!adcIsConversionComplete(adcREG1, adcGROUP1)); 
+
+	/** - Read the conversion result
+	*     The data contains the Light sensor data
+    */
+	adcGetData(adcREG1, adcGROUP1, &lightData);
+	
+	/** - Transmit the Conversion data to PC using SCI
+    */
+	return lightData.value;
+}
+
 static void lightServiceTask(void * pvParameters) {
     /* USER CODE BEGIN */
     // Wait for MEASURE_LIGHT event in the queue and then print the ambient light value to the serial port.
+    ASSERT(lightServiceTaskHandle != NULL);
+    light_event_t lightEvent = NULL_EVENT;
+	
+        if (xQueueReceive(xLightServiceQueue, &lightEvent, portMAX_DELAY) == pdTRUE) {
+		
+            if (lightEvent == MEASURE_LIGHT){
+            
+            	uint16_t lightSensorData = getLightSensorData();
+		int dataSize = (int)(ceil(log(lightSensorData))+1);
+            	char data[dataSize];
```

Comment:
You can't use a non-constant variable to declare an array like this. If you wanted to have a variable-size array, you'd use `malloc`. However, we don't use `malloc` because it introduces some non-deterministic behaviour. Instead, the size of the array should be known at compile time. I.e. `dataSize` must be a constant value. You can just make it big enough to store a 4-digit number with the `\r\n` characters. So a size of 6 is enough. 

Since we don't use magic numbers, you should declare a macro for the size or use a constant variable:
```
#define LIGHT_DATA_STR_LEN 6U
...
char data[LIGHT_DATA_STR_LEN];
```
or
```
const uint8_t lightDataStrLen = 6;
char data[lightDataStrLen];
```